anomalib/models/components/base/anomaly_module.py

```diff
@@ -36,30 +35,41 @@ class AnomalyModule(pl.LightningModule, ABC):
     Acts as a base class for all the Anomaly Modules in the library.
 
     Args:
-        params (Union[DictConfig, ListConfig]): Configuration
+        adaptive_threshold (Optional[bool], optional): Boolean to automatically choose adaptive threshold.
+            Defaults to None.
+        default_image_threshold (Optional[float], optional): Manual default image threshold.
+            Defaults to None.
+        default_pixel_threshold (Optional[float], optional): Manaul default pixel threshold.
+            Defaults to None.
     """
 
-    def __init__(self, params: Union[DictConfig, ListConfig]):
-
+    def __init__(
+        self,
+        adaptive_threshold: Optional[bool] = None,
+        default_image_threshold: Optional[float] = None,
+        default_pixel_threshold: Optional[float] = None,
```

Comment:
Would it be an idea to initialize these in the metrics callback instead of here?